SetupDataPkg/Tools/VariableList.py

```diff
@@ -1278,6 +1277,68 @@ def write_csv(schema, csv_path, full, subknobs=True):
                             knob.format.object_to_string(knob.value),
                             string_binary,
                             knob.help])
+                            
+                            
+def write_csv_detailed(schema, csv_path):
+
+    with open(csv_path, 'w', newline='') as csv_file:
+        writer = csv.writer(csv_file)
+        guid = None
+        namespace_changed = True # Used to print the guid on the first row and then only print
+                                 # another guid if it changes
+        
+        writer.writerow(['Guid', 'Knob', 'Value', 'Binary', 'Default Value', 'Valid Value Options', 'Help'])
+        # Get the knobs one at a time
+        for knob in schema.knobs:
+            # Check if the guid has changed
+            if knob.namespace == guid:
+                namespace_changed = False
+            else:
+                guid = knob.namespace
+                namespace_changed = True
+                
+            # For each subknob of the current knob, if it's a leaf, write it to the CSV file
+            for subknob in knob.subknobs:  
+                subknob_values = ""
+                
+                # Get the binary value
+                binary = subknob.format.object_to_binary(subknob.value)
+                string_binary = " ".join(map("%2.2x".__mod__, binary))
+                # Check if subknob is a leaf; skip otherwise
+                if subknob.leaf:
+                    # Get all valid values for each knob
+                    for enum in schema.enums:

```

Comment:
You shouldn't need to loop to search for the type name. I believe "subknob.format" is a reference directly to the "format" object.

I think you want something like this:

if subknob.format is EnumFormat:
    subknob_values = subknob.format.values;
elif subknob.format is BoolFormat:
   subknob_values = [ "TRUE", "FALSE" ]

Reply:
thank you for the suggestion. Changes applied.